### PR TITLE
ExtensionDeploymentStatus new Actions

### LINKED ISSUE #977: [BC Idea]: Action in Page 2508 to Extension Management
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=7a60b923-4f01-ef11-a73c-0022484ddea9

### Description

When uploading a PTE we have to refresh the "Extension Installation Status" page, which the is the main page in the browser. When we want to navigate back to the "installed extensions" page you have to search again.

the idea is a simple navigation button to open the page "2500 "Extension management"

### I will provide the implementation for this BC Idea

- [X] I will provide the implementation for this BC Idea
Internal work item: [AB#527490](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/527490)

### PR DESCRIPTION
**Pull request by JulianTillmann**
#### Summary
Added new actions to page "Extensions Deployment Status"

- Refresh: for refreshing the status of the deployment
- Upload Extension: to re-upload an extension if the first try failed for some reason
- Extension Management: to go back to the extension management page

Unfortunately it's not possible to test this Onprem and there does not seem to be a whole lot of test automation for this which I could re-use. Hopefully you can qualitity check this manually.

#### Work Item(s)
Fixes #977

Fixes [AB#527490](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/527490)

### CODE REVIEW

**Comment by JesperSchulz** [comment]
Hi @JulianTillmann,

you've got a few easy to fix codecop violations. Could you look into those to get green ticks in the CI step? Then we'll process the PR 🙂

File: src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al [hunk 1]
```diff
@@ -99,6 +100,49 @@ page 2508 "Extension Deployment Status"
...
+                    ExtensionOperationImpl: Codeunit "Extension Operation Impl";
+                begin
+                    NavAppTenantOperationTable.CopyFilters(Rec);
+                    if NavAppTenantOperationTable.FindSet() then
+                        repeat
+                            ExtensionOperationImpl.RefreshStatus(NavAppTenantOperationTable."Operation ID");
+                        until NavAppTenantOperationTable.Next() = 0;
+                end;
+            }
+            action("Upload Extension")
```
Comment (EmilDamsbo) [inline src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al:122]: Everywhere else, we are adding the `ApplicationArea = All`
While it is not required, let's be consistent in this and add it here too.

**Review by EmilDamsbo — CHANGES_REQUESTED** [review]

File: src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al [hunk 2]
```diff
@@ -99,6 +100,49 @@ page 2508 "Extension Deployment Status"
...
+                ApplicationArea = All;
+                ToolTip = 'Refresh the deployment details.';
+                Image = Refresh;
+
+                trigger OnAction()
+                var
+                    NavAppTenantOperationTable: Record "NAV App Tenant Operation";
+                    ExtensionOperationImpl: Codeunit "Extension Operation Impl";
+                begin
+                    NavAppTenantOperationTable.CopyFilters(Rec);
```
Comment (BazookaMusic) [inline src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al]: I think this doesn't copy marked records (rows you select with cntrl)

This one should (although it also copies the value of the record)
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/record/record-copy-method

I

Comment (BazookaMusic) [reply src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al]: Without marked records, all the rows in view will be refreshed even if you select some

Comment (BazookaMusic) [reply src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al]: I might be wrong though and copy filters does copy the marked records, it should be straightforward to test by selecting rows and calling the action

File: src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al [hunk 3]
```diff
@@ -99,6 +100,49 @@ page 2508 "Extension Deployment Status"
...
+                    if NavAppTenantOperationTable.FindSet() then
+                        repeat
+                            ExtensionOperationImpl.RefreshStatus(NavAppTenantOperationTable."Operation ID");
```
Comment (thloke) [inline src/System Application/App/Extension Management/src/ExtensionDeploymentStatus.Page.al:118]: We only need to refresh status if the status is `Unknown`, `NotFound`, or `InProgress`. Once it's `Failed` or `Completed` it's not going to change anymore. This should cut down the number of operations that actually need refreshing. Might be easier still to just set a filter on `NavAppTenantOperationTable` before `FindSet` to get the records that aren't in `Failed` or `Completed` state.

**Review by EmilDamsbo — APPROVED** [review]

**Review by thloke — APPROVED** [review]

**Pull request merged**